apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -88,7 +90,31 @@ export class SendMessage {
       })
     );
 
-    const stepType = command.step?.template?.type;
+    if (!command._templateId) {
+      throw new PlatformException('Template ID is required');
+    }
+
+    const stepId = command.step._id;
+    const stepType = command.step.template?.type;
+
+    let stepTemplateResult;
+    if (command.step.template?.type !== StepTypeEnum.TRIGGER) {
+      if (!stepId) {
+        throw new PlatformException('Step ID is required');
+      }
+
+      stepTemplateResult = await this.stepTemplateFetcher.fetchStepTemplate({
+        workflowId: command._templateId,
+        stepId: stepId || '',
```

Comment:
strange did TS yield that it can be undefined after the if condition above?
```suggestion
        stepId,
```